pkg/registry/extensions/deployment/strategy.go

```diff
@@ -77,6 +77,12 @@ func (deploymentStrategy) PrepareForUpdate(ctx api.Context, obj, old runtime.Obj
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
 
+	if newDeployment.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
```

Comment:
Looking at this code, this seems to be something that ought to have been in validation code. This is not setting a default, this is fixing an error in user config. IMO this should be rejected instead of cleared.

Of course that would be a compatibility break and you'd have to figure out what exactly needs to be mitigated there.